### PR TITLE
Allows l10n_push script to push only a specific resource.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#9430

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] If adding a new directory with NPM packages, ensure this directory is
  added to `scripts/audit.js`.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mkarolin** [comment]
ping @petemill @bsclifton

File: lib/pullL10n.js [hunk 1]
```diff
@@ -17,7 +18,7 @@ const pullL10n = (options) => {
   }
 
   braveTopLevelPaths.forEach((sourceStringPath) => {
-    if (!options.grd_path || sourceStringPath.endsWith(`/${options.grd_path}`))
+    if (!options.grd_path || sourceStringPath.endsWith(path.sep + `${options.grd_path}`))
```
Comment (petemill) [inline lib/pullL10n.js:21]: Actually this should just be `path.sep + options.grd_path`

File: lib/pushL10n.js [hunk 2]
```diff
@@ -22,7 +23,8 @@ const pushL10n = (options) => {
     args = ['checkout', '--', '*.grd*']
     util.run('git', args, runOptions)
     braveTopLevelPaths.forEach((sourceStringPath) => {
-      util.run('python', ['script/push-l10n.py', '--source_string_path', sourceStringPath], cmdOptions)
+      if (!options.grd_path || sourceStringPath.endsWith(path.sep + `${options.grd_path}`))
```
Comment (petemill) [inline lib/pushL10n.js:26]: Actually this should just be `path.sep + options.grd_path`

File: scripts/commands.js [hunk 3]
```diff
@@ -128,6 +128,7 @@ program
 program
   .command('push_l10n')
   .option('--extension <extension>', 'Scope this command to localize a Brave extension such as ethereum-remote-client')
+  .option('--grd_path <grd_path>', `Relative path to match end of full GRD path, e.g: '/generated_resources.grd'.`)
```
Comment (petemill) [inline scripts/commands.js:131]: looks like we don't want to suggest `/` prefix

**Comment by mkarolin** [comment]
Thanks @petemill, updated.

**Comment by mkarolin** [comment]
Closing in favor of brave/brave-core#6294

**Pull request closed without merging**